sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -733,6 +724,33 @@ struct _pi_context : _pi_object {
   std::unordered_map<void *, MemAllocRecord> MemAllocs;
 
 private:
+  // Calculate root device.
+  pi_device getRootDevice() const {
+    if (Devices.size() == 1)
+      return Devices[0];
+
+    // Check if we have context with subdevices of the same device (context
+    // may include root device itself as well)
+    pi_device ContextRootDevice =
+        Devices[0]->RootDevice ? Devices[0]->RootDevice : Devices[0];
```

Comment:
maybe add a check for "0" devices too at least to keep static verifiers happy?

Reply:
Added check but please notice that this code was just moved from constructor